test/methods.js

```diff
@@ -618,6 +618,55 @@ test( "remote, highlight all invalid fields", function( assert ) {
 		done();
 	}, 500 );
 } );
+test( "remote, unhighlighted should be invoked after being highlighted/invalid", function( assert ) {
+	expect( 2 );
+
+	var done1 = assert.async(),
+		done2 = assert.async(),
+		$form = $( "#testForm25" ),
+		$somethingField = $form.find( "input[name='something25']" ),
+		responseText = "false",
+		response = function() { return responseText; },
+		validateOptions = {
+			highlight: function( e ) {
+				$( e ).addClass( "error" );
+			},
+			unhighlight: function( e ) {
+				$( e ).removeClass( "error" );
+			},
+	        rules: {
+				something25: {
+	                required: true,
+	                remote: {
+	                    url: "callback.php",
+	                    type: "post",
+						data: {
+							responseText: response
+						},
+						async: false
```

Comment:
The async: false was necessary in order to prove that the the test failed before and passed after the change in core.js. The test passes if it's removed, but not for the right reasons. The point of the change in core.js is to avoid resetting this.currentElements inside the success callback before validator.showErrors() is invoked.